osu.Game/Overlays/News/Sidebar/NewsSideBar.cs

```diff
@@ -0,0 +1,113 @@
+using osuTK;
+using System.Collections.Generic;
+
+namespace osu.Game.Overlays.News.Sidebar
+{
+    public class NewsSideBar : CompositeDrawable
+    {
+        public readonly Bindable<APINewsSidebar> Metadata = new Bindable<APINewsSidebar>();
+
+        private YearsPanel yearsPanel;
+        private FillFlowContainer<MonthPanel> monthsFlow;
+
+        [BackgroundDependencyLoader]
+        private void load(OverlayColourProvider colourProvider)
+        {
+            RelativeSizeAxes = Axes.Y;
+            Width = 250;
+            InternalChildren = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider.Background4
+                },
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Padding = new MarginPadding
+                    {
+                        Top = 20,
+                        Left = 50,
+                        Right = 30
+                    },
+                    Child = new FillFlowContainer
+                    {
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.TopCentre,
+                        Direction = FillDirection.Vertical,
+                        AutoSizeAxes = Axes.Both,
+                        Spacing = new Vector2(0, 20),
+                        Children = new Drawable[]
+                        {
+                            yearsPanel = new YearsPanel(),
+                            monthsFlow = new FillFlowContainer<MonthPanel>
+                            {
+                                AutoSizeAxes = Axes.Both,
+                                Direction = FillDirection.Vertical,
+                                Spacing = new Vector2(0, 10)
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Metadata.BindValueChanged(metadata =>
+            {
+                monthsFlow.Clear();
+
+                if (metadata.NewValue == null)
+                {
+                    yearsPanel.Hide();
+                    return;
+                }
+
+                yearsPanel.Years = metadata.NewValue.Years;
+                yearsPanel.Show();
+
+                if (metadata.NewValue != null)
+                {
+                    var dict = new Dictionary<int, List<APINewsPost>>();
+
+                    foreach (var p in metadata.NewValue.NewsPosts)
+                    {
+                        var month = p.PublishedAt.Month;
+
+                        if (dict.ContainsKey(month))
+                            dict[month].Add(p);
+                        else
+                        {
+                            dict.Add(month, new List<APINewsPost>(new[] { p }));
+                        }
```

Comment:
can be shortened using linq `.ToLookup(post => post.PublishedAt.Month)`

Reply:
was gonna suggest that, but then `post.PublishedAt.Year` comes into play, maybe a tuple could work if it supports equality properly.

Reply:
How are years relevant here? The articles shown should be from one year, as per web, no?

Reply:
Right, been confused about how this works.